src/Client.php

```diff
@@ -119,12 +119,8 @@ public function __construct(?string $uri = null, array $uriOptions = [], array $
             throw InvalidArgumentException::invalidType('"typeMap" driver option', $driverOptions['typeMap'], 'array');
         }
 
-        if (isset($driverOptions['autoEncryption']['keyVaultClient'])) {
-            if ($driverOptions['autoEncryption']['keyVaultClient'] instanceof self) {
-                $driverOptions['autoEncryption']['keyVaultClient'] = $driverOptions['autoEncryption']['keyVaultClient']->manager;
-            } elseif (! $driverOptions['autoEncryption']['keyVaultClient'] instanceof Manager) {
-                throw InvalidArgumentException::invalidType('"keyVaultClient" autoEncryption option', $driverOptions['autoEncryption']['keyVaultClient'], [self::class, Manager::class]);
-            }
+        if (isset($driverOptions['autoEncryption']) && is_array($driverOptions['autoEncryption'])) {
+            $driverOptions['autoEncryption'] = $this->formatEncryptionOptions($driverOptions['autoEncryption']);
```

Comment:
```suggestion
            $driverOptions['autoEncryption'] = $this->prepareAutoEncryptionOptions($driverOptions['autoEncryption']);
```

Leaving this suggestion in case you accept the other as-is.

Reply:
The method must be moved to the pedantic order.